demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+    """
+    returns the xdm type convention
+    """
+    converting_dict = {"string": "String", "int": "Number", "boolean": "Boolean"}
+
+    return converting_dict.get(schema_type, "String")
+
+
+def convert_to_xdm_type(name: str, xdm_type: str) -> str:
+    """
+    Wraps the xql with a conversion to fit the xdm schema if the raw response type is incompatible with the schema type
+    """
+    if xdm_type == "String":
+        name = to_string(name)
+    elif xdm_type == "Number":
+        name = to_number(name)
+
+    return name
+
+
+def create_xif_file(
+    mapping_list: List[MappingField], outputfile_xif: Path, dataset_name: str
+) -> None:
+    """
+    Created the xif file for the modeling rules
+    """
+    logger.info("Generating xif file\n")
+    xif_rule = create_xif_header(dataset_name)
+    for mapping_rule in mapping_list:
+        logger.info(
+            f'field name: "{mapping_rule.field_path_raw}" - xdm type: {mapping_rule.xdm_field_type} - raw type {mapping_rule.type_raw}'
+        )
+        name = mapping_rule.field_path_raw
+
+        if "." in mapping_rule.field_path_raw:
+            dict_keys = mapping_rule.field_path_raw.split(".")
+            prefix = dict_keys[0]
+            suffix = ".".join(dict_keys[1:])
+            if mapping_rule.xdm_class_type == "Array":
+                name = json_extract_array(prefix, suffix)
+                xif_rule += f"\t{mapping_rule.xdm_rule} = {name},\n"
+                continue
+            else:
+                name = json_extract_scalar(prefix, suffix)
+
+        if mapping_rule.xdm_field_type != convert_raw_type_to_xdm_type(
+            mapping_rule.type_raw
+        ):
+            # Tpue casting
+            name = convert_to_xdm_type(name, mapping_rule.xdm_field_type)
+
+        if mapping_rule.xdm_class_type == "Array":
+            # convert a scalar into an array
+            name = array_create(name)
+
+        xif_rule += f"\t{mapping_rule.xdm_rule} = {name},\n"
+
+    xif_rule = replace_last_char(xif_rule)
+
+    with open(outputfile_xif, "w") as f:
+        f.write(xif_rule)
+
+
+def replace_last_char(s: str) -> str:
+    """
+    Replaces the last char of the xif file to be ;
+    """
+    s = s[:-2]
+    s += ";\n"
+    return s
+
+
+def create_scheme_file(
+    mapping_list: List[MappingField], dataset_name, outputfile_schema
+):
+    """
+    Creates the .json schema file
+    """
+    logger.info("creating modeling rules schema\n")
+    name_type_dict = {}
+    for mapping_rule in mapping_list:
+        keys_list = mapping_rule.field_path_raw.split(".")
+        name = keys_list[0]
+        if name not in name_type_dict:
+            name_type_dict[name] = mapping_rule.create_schema_types()
+    modeling_rules_json = {dataset_name: name_type_dict}
+
+    with open(outputfile_schema, "w") as f:
+        res = json.dumps(modeling_rules_json, indent=4)
+        f.write(res)
+
+
+def process_yml_name(product: str, vendor: str):
```

Comment:
please add return type hints to all methods

Reply:
Done